### PR TITLE
breakout object bits into scoped enum

### PR DESCRIPTION
**Pull request by aronwk-aaron**
tested that things still work as expected
Waiting on #996 since there's likely to be merge conflicts due to includes, thanks git

### CODE REVIEW

File: dCommon/GeneralUtils.cpp [hunk 1]
```diff
@@ -202,18 +202,18 @@ bool GeneralUtils::CaseInsensitiveStringCompare(const std::string& a, const std:
...
+int64_t GeneralUtils::ClearBit(int64_t value, eObjectBits index) {
+	return value &= ~(1ULL << static_cast<uint32_t>(index));
 }
 
 //! Checks a specific bit in a signed 64-bit integer
-bool GeneralUtils::CheckBit(int64_t value, uint32_t index) {
-	return value & (1ULL << index);
+bool GeneralUtils::CheckBit(int64_t value, eObjectBits index) {
+	return value & (1ULL << static_cast<uint32_t>(index));
 }
```
Comment (aronwk-aaron) [inline dCommon/GeneralUtils.cpp:217]: Real question: could I avoid this static cast with an operator overload

Comment (EmosewaMC) [reply dCommon/GeneralUtils.cpp:217]: I may have an idea to alleviate these

**Review by EmosewaMC — APPROVED** [review]
lgtm

**Review by EmosewaMC — DISMISSED** [review]
lgtm maybe inline in the future since the function is small

**Review by EmosewaMC — APPROVED** [review]

**Review by Jettford — DISMISSED** [review]

**Review by EmosewaMC — APPROVED** [review]
lol

**Review by DarwinAnim8or — COMMENTED** [review]
While I'm not going to consider this as *blocking*, note that as per the CONTRIBUTING.md file, you should use "#pragma once" and not C-style header guards

**Review by DarwinAnim8or — APPROVED** [review]

**Comment by EmosewaMC** [comment]
> While I'm not going to consider this as _blocking_, note that as per the CONTRIBUTING.md file, you should use "#pragma once" and not C-style header guards

Pre compiled headers cannot use pragma once.  There are issues with pragma once which require standard include guards when pre compiled headers are used.

**Comment by DarwinAnim8or** [comment]
> 

Looks like it's a GCC thing? "that they have no plan to add at the present" 
🙃

**Review by EmosewaMC — APPROVED** [review]
4 merge conflicts later

**Pull request merged**